### PR TITLE
[Merged by Bors] - feat: port LinearAlgebra.GeneralLinearGroup

### PR DESCRIPTION
**Pull request by xroblot**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/LinearAlgebra/GeneralLinearGroup.lean [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+
+-- Porting note: This is not necessary anymore
+-- instance : CoeFun (GeneralLinearGroup R M) fun _ ↦ M → M := by infer_instance
+
+/-- An invertible linear map `f` determines an equivalence from `M` to itself. -/
+def toLinearEquiv (f : GeneralLinearGroup R M) : M ≃ₗ[R] M :=
+  { f.val with
+    invFun := f.inv.toFun
+    left_inv := fun m ↦ show (f.inv * f.val) m = m by erw [f.inv_val] ; simp
+    right_inv := fun m ↦ show (f.val * f.inv) m = m by erw [f.val_inv] ; simp }
```
Comment (jcommelin) [inline Mathlib/LinearAlgebra/GeneralLinearGroup.lean:51]:
```suggestion
    left_inv := fun m ↦ show (f.inv * f.val) m = m by erw [f.inv_val]; simp
    right_inv := fun m ↦ show (f.val * f.inv) m = m by erw [f.val_inv]; simp }
```

File: Mathlib/LinearAlgebra/GeneralLinearGroup.lean [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+variable (R M)
+
+/-- The general linear group on `R` and `M` is multiplicatively equivalent to the type of linear
+equivalences between `M` and itself. -/
+def generalLinearEquiv : GeneralLinearGroup R M ≃* M ≃ₗ[R] M where
+  toFun := toLinearEquiv
+  invFun := ofLinearEquiv
+  left_inv f := by ext ; rfl
+  right_inv f := by ext ; rfl
+  map_mul' x y := by ext ; rfl
```
Comment (jcommelin) [inline Mathlib/LinearAlgebra/GeneralLinearGroup.lean:71]:
```suggestion
  left_inv f := by ext; rfl
  right_inv f := by ext; rfl
  map_mul' x y := by ext; rfl
```

File: Mathlib/LinearAlgebra/GeneralLinearGroup.lean [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+#align linear_map.general_linear_group.general_linear_equiv
+  LinearMap.GeneralLinearGroup.generalLinearEquiv
+
+@[simp]
+theorem generalLinearEquiv_to_linearMap (f : GeneralLinearGroup R M) :
+    (generalLinearEquiv R M f : M →ₗ[R] M) = f := by ext ; rfl
```
Comment (jcommelin) [inline Mathlib/LinearAlgebra/GeneralLinearGroup.lean:77]:
```suggestion
    (generalLinearEquiv R M f : M →ₗ[R] M) = f := by ext; rfl
```

File: Mathlib/LinearAlgebra/GeneralLinearGroup.lean [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+#align linear_map.general_linear_group.general_linear_equiv_to_linear_map
+  LinearMap.GeneralLinearGroup.generalLinearEquiv_to_linearMap
```
Comment (jcommelin) [inline Mathlib/LinearAlgebra/GeneralLinearGroup.lean:79]:
```suggestion
#align linear_map.general_linear_group.general_linear_equiv_to_linear_map LinearMap.GeneralLinearGroup.generalLinearEquiv_to_linearMap
```

File: Mathlib/LinearAlgebra/GeneralLinearGroup.lean [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+
+@[simp]
+theorem coeFn_generalLinearEquiv (f : GeneralLinearGroup R M) :
+    (generalLinearEquiv R M f) = (f : M → M) := rfl
+#align linear_map.general_linear_group.coe_fn_general_linear_equiv
+  LinearMap.GeneralLinearGroup.coeFn_generalLinearEquiv
```
Comment (jcommelin) [inline Mathlib/LinearAlgebra/GeneralLinearGroup.lean:85]:
```suggestion
#align linear_map.general_linear_group.coe_fn_general_linear_equiv LinearMap.GeneralLinearGroup.coeFn_generalLinearEquiv
```

**Review by jcommelin — COMMENTED** [review]
Note that you can use the scripts `fix-lints.py` and `fix-comments.py` to help you a bit.

File: Mathlib/LinearAlgebra/GeneralLinearGroup.lean (hunk 1 above)
Comment (xroblot) [reply Mathlib/LinearAlgebra/GeneralLinearGroup.lean:51]: Thanks for the tip!

**Comment by ChrisHughes24** [comment]
bors merge

**Pull request closed without merging**